dwave/gate/circuit.py

```diff
@@ -355,7 +363,7 @@ def add_qregister(self, num_qubits: int = 0, label: Hashable = None) -> None:
         if label in self._qregisters:
             raise ValueError(f"Quantum register {label} already present in the circuit.")
 
-        data = [Qubit(str(i)) for i in range(num_qubits)]
+        data = [Qubit(str(i)) for i in range(self.num_qubits, self.num_qubits + num_qubits)]
```

Comment:
`QuantumRegister` only needs an iterable, not a sequence. So probably better to make `data` a generator rather than a list.